### PR TITLE
fix: issue where importing more constrained version causes failure

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

* **Fixes** issue where if I have file like:

```solidity
pragma solidity ^0.8.4;

// The file SpecificVersion.sol has pragma spec =0.8.12.
import "SpecificVersion.sol";

contract ImportConstrainingVersion {
    function foo() pure public returns(bool) {
        return true;
    }
}
```

It would fail to compile because it tries to use `0.8.14` to compile the import as well

* **Also fixes** bug where we would fail to process the NpmSpec for `pragma solidity >= 0.4.19 < 0.7.0;` because of the spaces between the operator and the version part (the proper way is `pragma solidity >=0.4.19 <0.7.0;`). But, now Ape will account for this.

CHANGELOG updates:

* fix: issue with projects using multiple solidity versions and imports
* fix: issue causing NpmSpec not to be processed when there are spaces between operator and number.
* fix: schema validation error when using the ABI Encoder V2 pragma
* fix: skip contracts with library linking found in their bytecode to prevent validation error

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by unparalleled-js** [comment]
- [ ] Found new issue with Ranged Pragmas

**Comment by unparalleled-js** [comment]
- [ ] Seems to be a regression this PR I found while trying to compile Open Zeppelin... Going to have to dig in to that...

**Review by fubuloubu — APPROVED** [review]

**Comment by unparalleled-js** [comment]
It appears I have broken the `chainlink-brownie-contracts` from compiling again... Stamdby as I attempt to fix.

File: ape_solidity/compiler.py [hunk 1]
```diff
@@ -270,68 +333,80 @@ def parse_contract_name(value: str) -> Tuple[Path, str]:
...
-                contract_type["runtimeBytecode"] = {"bytecode": contract_type["bin-runtime"]}
-                contract_type["userdoc"] = load_dict(contract_type["userdoc"])
-                contract_type["devdoc"] = load_dict(contract_type["devdoc"])
+                deployment_bytecode = contract_type["bin"]
+                runtime_bytecode = contract_type["bin"]
+
+                # Skip library linking.
+                if "$__" in deployment_bytecode or "__$" in runtime_bytecode:
+                    logger.warning("Libraries must be deployed and configured separately.")
+                    continue
```
Comment (fubuloubu) [inline ape_solidity/compiler.py:352]: Perfect, now just need config option for specifying libraries to pass through to `solc` and link

Comment (fubuloubu) [reply ape_solidity/compiler.py:352]: One sneaky thing that can be done here I guess is that if you are on a local network and specify `auto`, have it automatically deploy any libraries using a test_account and add that to the config somehow.

Absolutely atrocious I know, but so is this `solc` feature

Comment (unparalleled-js) [reply ape_solidity/compiler.py:352]: I created a separate issue for that since this PR is getting out-of-hand

Comment (unparalleled-js) [reply ape_solidity/compiler.py:352]: We are at least able to compile `yearn-vaults` AND `chainlink-brownie-projects` now!

**Comment by unparalleled-js** [comment]
Ugh... May have found some more issues with that chainlink repo again

File: ape_solidity/compiler.py [hunk 2]
```diff
@@ -163,8 +172,8 @@ def get_import_remapping(self, base_path: Optional[Path] = None) -> Dict[str, st
                 if not cached_manifest_file.exists():
                     raise CompilerError(f"Unable to find dependency '{suffix}'.")
 
-                manifest_dict = json.loads(cached_manifest_file.read_text())
-                manifest = PackageManifest(**manifest_dict)
+                # NOTE: Purposely skip Pydantic validation here.
+                manifest = PackageManifest(**json.loads(cached_manifest_file.read_text()))
```
Comment (fubuloubu) [inline ape_solidity/compiler.py:176]: I think this still does validation, the suggestion from the docs is to use `.construct`
https://pydantic-docs.helpmanual.io/usage/models/#creating-models-without-validation

**Review by fubuloubu — APPROVED** [review]

File: ape_solidity/compiler.py (hunk 2 above)
Comment (unparalleled-js) [reply ape_solidity/compiler.py:176]: weird, it doesn't seem like it does! But good to know

Comment (unparalleled-js) [reply ape_solidity/compiler.py]: Done!

Comment (unparalleled-js) [reply ape_solidity/compiler.py:175]: Welp, so this causes the `Source` objects to never get serialized as classes and that causes mypy issues, so I reverted this.

There seems to be a difference between using the normal `init` versus `parse_obj()` where the init still gives me classes but does not run validators

**Pull request merged**